### PR TITLE
feat(android): BREAKING CHANGE upgrading to sdk 28 and androidx (#2068)

### PR DESCRIPTION
**Pull request by n1ru4l**
# Summary

Cherry Pick of #2068 which was reverted in #2305

Before merging this we need a correct migration path as mentioned in #2296

## Known problems

- https://github.com/react-native-community/react-native-camera/issues/2296#issuecomment-499599760
- #2302
- #2301

cc @sibelius @banshee @gla5001 @kassim

@mikehardy It seems like you would know what has to be done before merging this? Would you have the capacity for getting this pull request ready for merging?

### CODE REVIEW

**Comment by mikehardy** [comment]
It will depend on this: https://github.com/react-native-community/discussions-and-proposals/issues/129

We currently have an "ok" plan (generate AARs at library build time / pre-publish) and are testing/iterating towards what we hope is a "great" plan (do a jetify transform at project build time). It may be a little while (week? 2 weeks?) but that issue will have the status

**Comment by sibelius** [comment]
@n1ru4l @mikehardy can we merge this now https://facebook.github.io/react-native/blog/2019/07/03/version-60?

**Comment by mikehardy** [comment]
Oh sure! Jetify -r should help users that can't move yet. Check maintainers section on jetifier for gradle changes just in case but should be fine?

**Comment by n1ru4l** [comment]
Please check semantic-release commit message format for breaking changes when merging. Adding some info about jetify to the install instructions would also be a good thing to do before merging

**Comment by mrigo** [comment]
Hello,
any news about supporting React Native 0.60 in Android?
Thanks!

**Comment by sibelius** [comment]
@mrigo can you use this branch/pr and make sure this work properly

when more developers says it is all good, we gonna merge this

**Comment by mrigo** [comment]
Thanks!
I've just tried and it seems working!

**Comment by sibelius** [comment]
does autolink works well?

**Comment by n1ru4l** [comment]
The commit message was not in a valid angular commit message syntax, therefore no build was triggered.

https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#footer

This would have been correct:

```
feat(android): upgrade android sdk to 28 and androidx

BREAKING CHANGE: users that use react-native < 0.60.0 will have to use jetifier (https://github.com/mikehardy/jetifier)
```

**Comment by VeirryAu** [comment]
how to use your version @n1ru4l in my package.json ?

**Comment by VeirryAu** [comment]
I've trying to change react-native-camera like the way you change it @n1ru4l . It works fine for me.

**Comment by VeirryAu** [comment]
@sibelius is this pull request will merge ?

**Comment by sibelius** [comment]
I've used the wrong commit message

so we need to try to do another release

@n1ru4l can you help us?

File: android/build.gradle [hunk 1]
```diff
@@ -81,9 +76,9 @@ dependencies {
   implementation "com.google.zxing:core:3.3.3"
   implementation "com.drewnoakes:metadata-extractor:2.11.0"
   generalImplementation "com.google.android.gms:play-services-vision:$googlePlayServicesVisionVersion"
-  implementation "com.android.support:exifinterface:${safeExtGet('supportLibVersion', '28.0.0')}"
-  implementation "com.android.support:support-annotations:${safeExtGet('supportLibVersion', '28.0.0')}"
-  implementation "com.android.support:support-v4:${safeExtGet('supportLibVersion', '28.0.0')}"
+  implementation "androidx.exifinterface:exifinterface:1.0.0"
```
Comment (ronal2do) [inline android/build.gradle]: maybe is better to move it to `safeExtGet` to, androidXSupportVersion

**Comment by n1ru4l** [comment]
@sibelius revert and merge again with correct message 😅

**Comment by sibelius** [comment]
Can’t we bump directly on master?

**Comment by n1ru4l** [comment]
@Sibelius Push an empty commit to master with the breaking change commit message

**Comment by pinguo-zhouwei** [comment]
RN0.55, and my project has been migrated to androidx,  which version of react-native-camera should i use ?

**Pull request merged**